src/main/java/com/cegeka/academy/service/challengeAnswer/ChallengeAnswerServiceImp.java

```diff
@@ -80,4 +101,38 @@ public void deleteChallengeAnswer(Long userId, Long challengeId) throws NotFound
         logger.info("Challenge answer was deleted.");
 
     }
+
+    @Override
+    public void uploadAnswer(Long userId, Long challengeId, ChallengeAnswerDTO challengeAnswerDTO) throws IOException, NotFoundException {
+
+        userRepository.findById(userId).orElseThrow(() -> new NotFoundException().setMessage("User not found"));
+        challengeRepository.findById(challengeId).orElseThrow(() -> new NotFoundException().setMessage("Challenge not found"));
+        UserChallenge userChallenge = userChallengeRepository.findByUserIdAndChallengeId(userId, challengeId)
+                .orElseThrow(() -> new NotFoundException().setMessage("User challenge not found"));
+
```

Comment:
Daca gasesti Userchallenge dupa userId si challengeId atunci nu este clar ca exista in baza de date? De ce mai faci userRepo.findById si challengeRepo.findById? Este redundant